### DISCUSSION TITLE
Multiprocessing of Appium drivers to run multiple app instances

### DISCUSSION THREAD

**Discussion by MagdelineNg** [appium/appium#19633] [Q&A]
I want to launch 3 instances of the same app with same bundleId, and execute actions on each sequentially (for eg. I want to login on one instance, then check for the login status on the second and third app instances) . Would multiprocessing or using multiple threads be better? I have a minimal reproducible code for multiprocessing [here](https://discuss.appium.io/t/multiprocessing-for-different-app-instances/41323), but I am facing an error

**Comment by jlipps** [comment]
Do you mean on separate devices, with three separate Appium sessions? You probably don't need multiprocessing. Just do something like this:

- start session 1
- start session 2
- start session 3
- use session 1 to login
- use session 2 to verify login
- use session 3 to verify login
- close all sessions

**Reply by MagdelineNg** [reply]
Hi, are you suggesting launching 3 different drivers for each separate session? I used activateApp to launch different sessions of the same app (same bundle ID) using one driver, but encountered some errors. Here is the code used: https://discuss.appium.io/t/create-multiple-instances-of-the-same-app/41266

**Reply by jlipps** [reply]
You're trying to open multiple instances of the same app on the same device? That's probably not possible. How would a user do this?

**Reply by MagdelineNg** [reply]
@jlipps Yes, I want to start multiple instances of the same app to check whether actions in one instance will affect the other app instances. To do this in command line, I can just use ```open -n /Desktop/MyMacOSapp.app```.

**Reply by jlipps** [reply]
@mykola-mokhnach any ideas here? not sure how we do activateApp on the mac driver; maybe it's not possible to launch multiple instances of the app that way?

**Reply by jlipps** [reply]
closing as duplicate of https://github.com/appium/appium-mac2-driver/issues/269

**Reply by mykola-mokhnach** [reply]
I have replied in https://discuss.appium.io/t/create-multiple-instances-of-the-same-app/41266/8
